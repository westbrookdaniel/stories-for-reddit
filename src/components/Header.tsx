import React from 'react'
import { Link } from 'react-router-dom'
import { HStack } from '@chakra-ui/core'

export default function Header() {
    return (
        <HStack as="nav" spacing={10}>
           <Link to='/'>Home</Link>
           <Link to='/about'>About</Link>
        </HStack>
    )
}
