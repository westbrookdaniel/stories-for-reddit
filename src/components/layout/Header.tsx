import React from 'react'
import { Link } from 'react-router-dom'
import { HStack, Button, useColorMode, Box, Heading } from '@chakra-ui/core'
import { FcCdLogo, FcDvdLogo } from 'react-icons/fc'
import Logo from '../General/Logo'

export default function Header() {
	const { toggleColorMode, colorMode } = useColorMode()

	return (
		<Box d="flex" alignItems="center" w="100%">
            <Logo colorMode={colorMode} fontSize="8rem" />
            <Heading as="h2" fontSize="1.2rem" w="100%" ml={8}>Stories for Reddit</Heading>
			<HStack as="nav" justifyContent="flex-end" w="100%" spacing={10}>
				<Link to="/">Home</Link>
				<Link to="/style-guide">Style Guide</Link>
				<Button onClick={toggleColorMode}>Color Mode</Button>
			</HStack>
		</Box>
	)
}
