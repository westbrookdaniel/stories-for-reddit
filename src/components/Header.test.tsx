import React from 'react'
import Header from './Header'
import { Router } from 'react-router-dom'
import { render, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import { createMemoryHistory } from 'history'

afterEach(cleanup)

describe('Header', () => {
	const history = createMemoryHistory()
	const route = '/'
	history.push(route)

	it('should render', () => {
		const header = render(
			<Router history={history}>
				<Header />
			</Router>
		)

		expect(header).toMatchSnapshot()
	})

	it('should not break when using links', () => {
		const header = render(
			<Router history={history}>
				<Header />
			</Router>
		)

		fireEvent.click(header.getByText(/home/i))
		fireEvent.click(header.getByText(/about/i))

		expect(header).toMatchSnapshot()
	})
})
