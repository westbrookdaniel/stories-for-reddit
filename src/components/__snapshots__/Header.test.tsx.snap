// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should not break when using links 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <span
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; margin: 0px auto; text-align: center; top: 0px; right: 0px; left: 0px;"
      />
      <span
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; top: 0px; left: 0px;"
      />
      <span
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; top: 0px; right: 0px;"
      />
      <span
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; bottom: 0px; left: 0px;"
      />
      <span
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; margin: 0px auto; text-align: center; bottom: 0px; right: 0px; left: 0px;"
      />
      <span
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <nav
        class="chakra-stack css-1hox38z"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/about"
        >
          About
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="chakra-stack css-1hox38z"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about"
      >
        About
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <span
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; margin: 0px auto; text-align: center; top: 0px; right: 0px; left: 0px;"
      />
      <span
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; top: 0px; left: 0px;"
      />
      <span
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; top: 0px; right: 0px;"
      />
      <span
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; bottom: 0px; left: 0px;"
      />
      <span
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; margin: 0px auto; text-align: center; bottom: 0px; right: 0px; left: 0px;"
      />
      <span
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <nav
        class="chakra-stack css-1hox38z"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/about"
        >
          About
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="chakra-stack css-1hox38z"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about"
      >
        About
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
